torch/storage.py

```diff
@@ -329,6 +332,15 @@ def is_cuda(self):
     def is_hpu(self):
         return self.device.type == 'hpu'
 
+    @property
+    def filename(self) -> _Optional[str]:
+        """Returns the file name associated with this storage if the storage was memory mapped from a file.
+           or ``None`` if the storage was not created by memory mapping a file."""
+        if self._filename is None:
```

Comment:
This is a bit confusing.
I think you should either:
- Have a commend here explaining what None and -1 means for `._filename()`
- Always call `_get_filename()` as it is a cheap function anyways.

Reply:
Did the second!